### PR TITLE
LOG4J2-2707: ArrayIndexOutOfBoundsException in UuidUtil, when MAC add…

### PR DESCRIPTION
**Pull request by supertomcat**
LOG4J2-2707: ArrayIndexOutOfBoundsException in UuidUtil, when MAC address is longer than 6 bytes
https://issues.apache.org/jira/browse/LOG4J2-2707

### CODE REVIEW

**Comment by garydgregory** [comment]
You'll need some kind of unit test to show what was broken and is now fixed.

**Comment by supertomcat** [comment]
> 
> 
> You'll need some kind of unit test to show what was broken and is now fixed.

I'd love to provide one, but I wouldn't know how. To create a unit test for this case, I'd have to be able to mock NetUtils.getMacAddress() and return a longer than six bytes mac address. But that is not possible, because NetUtils is a final Utility class.

I documented in the Jira issue what exactly was broken and will be fixed by this Pull-Request. I hope this will be enough to get this Pull-Request merged.

**Comment by rgoers** [comment]
I will review this in the next few days.

**Comment by garydgregory** [comment]
> I'd love to provide one, but I wouldn't know how.

Simple: Refactor the static initializer into a **package private static** method that takes a byte array. Call that byte array from a new test.

**Comment by supertomcat** [comment]
> 
> 
> > I'd love to provide one, but I wouldn't know how.
> 
> Simple: Refactor the static initializer into a **package private static** method that takes a byte array. Call that byte array from a new test.

I refactored and added a unit test. I hope I did it, how you meant it.

File: log4j-core/src/main/java/org/apache/logging/log4j/core/util/UuidUtil.java [hunk 1]
```diff
@@ -65,6 +65,20 @@
 
     static {
         byte[] mac = NetUtils.getMacAddress();
+        LEAST = initialize(mac);
+    }
```
Comment (carterkozak) [inline log4j-core/src/main/java/org/apache/logging/log4j/core/util/UuidUtil.java]: This looks much better, thanks! Any chance we can replace the static block entirely?

```java
private static final long LEAST = initialize(NetUtils.getMacAddress());
```

Comment (supertomcat) [reply log4j-core/src/main/java/org/apache/logging/log4j/core/util/UuidUtil.java]: Done.

**Comment by rgoers** [comment]
Do NOT merge this. I need to review it.

**Pull request merged**